### ISSUE TITLE
$KITTY_LISTEN_ON not set! when using nnn with kitty

### ISSUE THREAD

**Issue by chenerg** [kovidgoyal/kitty#5117]
I'm using nnn with preview-tui plugin for preview images, however, when I toggle preview-tui, it says $KITTY_LISTEN_ON not set! 

My kitty.conf is 
`macos_option_as_alt yes
allow_remote_control yes
listen_on "unix:$TMPDIR/kitty"
`
The env related to kitty is 
`KITTY_WINDOW_ID=1
TERM=xterm-kitty
KITTY_PID=43716
TERMINFO=/Applications/kitty.app/Contents/Resources/kitty/terminfo
KITTY_INSTALLATION_DIR=/Applications/kitty.app/Contents/Resources/kitty
`
there seems to be no KITTY_LISTEN_ON.

Also I use Mac and I have tried ranger with nnn, which can use kitty for preview well.

**Comment by kovidgoyal** [comment]
You need to enable remote control in kitty.conf

**Comment by chenerg** [comment]
I have set `allow_remote_control yes` and `listen_on "unix:$TMPDIR/kitty"`, is there anything missing?

**Comment by chenerg** [comment]
Now, if I use `kitty --listen-on="unix:$TMPDIR/kitty"`, it works well in new window. But I have to open a terminal to open kitty in this way. My OS is macOS, I think it's `Unix`, why doesn't `listen_on "unix:$TMPDIR/kitty"` in kitty.conf work?

**Comment by kovidgoyal** [comment]
I cannot reproduce with

1) kitty --config NONE -o allow_remote_control=yes -o 'listen_on unix:/tmp/test'
2) env | grep KITTY_LIST
KITTY_LISTEN_ON=unix:/tmp/test-28907

**Comment by kovidgoyal** [comment]
Why do you think $TMPDIR is going to be defined by macOS in launch
services, which is where you are launching kitty from.

**Comment by chenerg** [comment]
Now it works! I think the main difference is changing from `listen_on "unix:$TMPDIR/kitty"` to `listen_on unix:$TMPDIR/kitty`.  Before that, I run `kitty --listen-on="unix:$TMPDIR/kitty"` and `kitty --listen-on=unix:$TMPDIR/kitty`,  both work well.

Then I change from `listen_on "unix:$TMPDIR/kitty"` to `listen_on unix:$TMPDIR/kitty`, restart, everything is fine now!

To reproduce the problem, I changed back to `listen_on "unix:$TMPDIR/kitty"`, unfortunately, kitty application cannot start this time. Then I change it back, now everything works FINE again!

Thank you very much, the above are what I conducted  before it working well.
